File: src/main/java/io/sc3/plethora/integration/computercraft/registry/ComputerCraftMetaRegistration.java
package pw.switchcraft.plethora.integration.computercraft.registry;

import dan200.computercraft.api.ComputerCraftAPI;
import net.minecraft.item.ItemStack;
import pw.switchcraft.plethora.api.meta.IMetaProvider;
import pw.switchcraft.plethora.api.meta.IMetaRegistry;
import pw.switchcraft.plethora.integration.computercraft.meta.item.*;

public class ComputerCraftMetaRegistration {
    public static void registerMetaProviders(IMetaRegistry r) {
        provider(r, "computer", ItemStack.class, new ComputerItemMeta());
        provider(r, "media", ItemStack.class, new MediaItemMeta());
        provider(r, "pocket", ItemStack.class, new PocketComputerItemMeta());
        provider(r, "printout", ItemStack.class, new PrintoutItemMeta());
        provider(r, "turtle", ItemStack.class, new TurtleItemMeta());
    }

    private static <T> void provider(IMetaRegistry r, String name, Class<T> target, IMetaProvider<T> provider) {
        r.registerMetaProvider(ComputerCraftAPI.MOD_ID + ":" + name, ComputerCraftAPI.MOD_ID, target, provider);
    }
}

File: src/main/java/io/sc3/plethora/integration/computercraft/registry/ComputerCraftMetaRegistration.kt
package io.sc3.plethora.integration.computercraft.registry

import dan200.computercraft.api.ComputerCraftAPI
import io.sc3.plethora.api.meta.IMetaProvider
import io.sc3.plethora.api.meta.IMetaRegistry
import io.sc3.plethora.integration.computercraft.meta.item.*

object ComputerCraftMetaRegistration {
  @JvmStatic
  fun registerMetaProviders(registry: IMetaRegistry) {
    with (registry) {
      provider("computer", ComputerItemMeta())
      provider("media", MediaItemMeta())
      provider("pocket", PocketComputerItemMeta())
      provider("printout", PrintoutItemMeta())
      provider("turtle", TurtleItemMeta())
    }
  }

  private inline fun <reified T> IMetaRegistry.provider(name: String, provider: IMetaProvider<T>) {
    registerMetaProvider("${ComputerCraftAPI.MOD_ID}:$name", ComputerCraftAPI.MOD_ID, T::class.java, provider)
  }
}
